crates/store/src/state.rs

```diff
@@ -170,8 +170,13 @@ impl State {
             .await?
             .ok_or(ApplyBlockError::DbBlockHeaderEmpty)?;
 
-        if block_num != prev_block.block_num() + 1 {
-            return Err(InvalidBlockError::NewBlockInvalidBlockNum.into());
+        let expected_block_num = prev_block.block_num() + 1;
```

Comment:
I believe there is now actually a `fn child` (or something like that).

```suggestion
        let expected_block_num = prev_block.block_num().child();
```